packages/block-library/src/image/view.js

```diff
@@ -411,7 +422,10 @@ const { state, actions, callbacks } = store(
 			initTriggerButton() {
 				const ctx = getContext();
 				const { ref } = getElement();
-				ctx.buttonRef = ref;
+				state.metadata[ ctx.imageId ] = {
+					...state.metadata[ ctx.imageId ],
+					buttonRef: ref,
+				};
```

Comment:
And here.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/63348/commits/3c3c3f27ea50e8c8749b8bd2e41a47afdc598142.